app/Config.hs

```diff
@@ -0,0 +1,231 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+module Config where
+
+import Prelude
+
+import Cardano.Metadata.GoguenRegistry
+    ( GoguenRegistryEntry (..), PartialGoguenRegistryEntry )
+import Cardano.Metadata.Types
+    ( Subject (..), WellKnownProperty (..), emptyAttested )
+import Cardano.Slotting.Slot
+    ( SlotNo (..) )
+import Colog
+    ( pattern D, pattern E, pattern I, Severity, pattern W )
+import Control.Applicative
+    ( empty, optional, (<|>) )
+import Data.Foldable
+    ( asum )
+import Data.List
+    ( isSuffixOf )
+import Data.Text
+    ( Text )
+import Data.Void
+    ( Void )
+import qualified Options.Applicative as OA
+
+import qualified Data.Aeson as Aeson
+import qualified Data.Aeson.Types as Aeson
+import qualified Data.Bifunctor as Bifunctor
+import qualified Data.Text as T
+import qualified Text.Megaparsec as P
+import qualified Text.Megaparsec.Char as P
+import qualified Text.Megaparsec.Char.Lexer as P
+
+type Parser = P.Parsec Void Text
+
+data SlotNoPreference
+  = MainnetTip
+  | TestnetTip
+  | CustomSlot SlotNo
+  deriving (Eq, Show)
```

Comment:
I am afraid that is now obsolete ? We used to need the chain tip when validating time slots, yet this has been removed to favor useful use-cases with auto-expiring assets.

Reply:
Yes, this can be removed :+1: